internal/hcs/vmcompute.go

```diff
@@ -0,0 +1,549 @@
+package hcs
+
+import (
+	gcontext "context"
+	"syscall"
+
+	"github.com/Microsoft/hcsshim/internal/logfields"
+	"github.com/Microsoft/hcsshim/internal/timeout"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+const (
+	stuckSyscallWarning = "Syscall did not complete within operation timeout. This may indicate a platform issue. If it appears to be making no forward progress, obtain the stacks and see if there is a syscall stuck in the platform API for a significant length of time."
+)
+
+func hcsEnumerateComputeSystemsContext(ctx gcontext.Context, query string, computeSystems **uint16, result **uint16) (hr error) {
+	ctx, span := trace.StartSpan(ctx, "HcsEnumerateComputeSystems")
+	defer span.End()
+	span.AddAttributes(trace.StringAttribute("query", query))
+
+	ctx, cancel := gcontext.WithTimeout(ctx, timeout.SyscallWatcher)
+	defer cancel()
+
+	done := make(chan error, 1)
+	go func() {
+		done <- hcsEnumerateComputeSystems(query, computeSystems, result)
+	}()
+	select {
+	case <-ctx.Done():
+		if ctx.Err() == gcontext.DeadlineExceeded {
+			logrus.WithField(logfields.Timeout, timeout.SyscallWatcher).Warning(stuckSyscallWarning)
```

Comment:
We should use `log.G(ctx)` here.

Reply:
This was introduced in the oc PR :). I can move it here. I agree we want it in this case.

Reply:
Turns out I was wrong. Just didn't resolve for some reason. I moved to `log.G(ctx)`